### PR TITLE
feat(client): add show solution button in TimeLine

### LINKED ISSUE #38918: Add a Show Solution button for certification projects on the timeline
Currently when you complete a project it's shown on the timeline, but there isn't an easy way to see the solution that was submitted:

image

We should add a "Show Solution" button between the link to the project's page on the left and the date on the right. The button can be similar to the on the settings page, and it should open the solution that was submitted.

However, we'll need to handle the solutions for the JS Data Structures and Algorithm projects differently. Since the solutions are stored in the DB, maybe we can seed the solution back to the editor when the link in clicked. Or showing the submitted code in a popup modal might also be an option.

### PR DESCRIPTION
**Pull request by dipayanDebTheCoder**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38918

### CODE REVIEW

**Comment by raisedadead** [comment]
Nice work. Can we possibly add/update tests?

**Comment by dipayanDebTheCoder** [comment]
> Nice work. Can we possibly add/update tests?

Thanks for the reply. I believe Jest is being used to test the client portion of the Web Application. I read some of the code for the tests , and based on what I understood , tests could be added to check if the buttons with the respective links are being created for the timeline component, if that is helpful.
Also , I noticed that there is similar functionality in the **CertificationSettings component , but there are no tests written for it**.
If its required then I would be happy to add tests for both TimelIne and CertificationSettings components. But I believe it would be better if that would be in a separate issue with a PR of its own.

I apologize if I am suggesting anything wrong. I am very new to redux , gatsby , node.js, jest and contributing to opensource.

File: client/src/components/profile/components/TimeLine.js [hunk 1]
```diff
@@ -76,11 +84,69 @@ class TimelineInner extends Component {
     this.prevPage = this.prevPage.bind(this);
     this.nextPage = this.nextPage.bind(this);
     this.lastPage = this.lastPage.bind(this);
+    this.renderViewButton = this.renderViewButton.bind(this);
+  }
+
+  renderViewButton(Id, files, githubLink, solution) {
```
Comment (huyenltnguyen) [inline client/src/components/profile/components/TimeLine.js:90]: I'm nit-picking here, could we follow the naming convention and change `Id` to `id`? :slightly_smiling_face:

Comment (dipayanDebTheCoder) [reply client/src/components/profile/components/TimeLine.js:90]: Sure thing.  I will change Id to id and push another commit. 

What should I do about the tests ? . Should I do a separate PR for the tests of TimeLine and CertificateSettings  after this gets merged or just include the tests for TimeLine along with the change to Id to id in a single commit in this PR itself ?

Thanks in advance

Comment (raisedadead) [reply client/src/components/profile/components/TimeLine.js:90]: You can expand the tests for bot timeline and certification settings in here. We have traditionally been lacking in tests, the more the merrier. Let us know if you are stuck.

Comment (dipayanDebTheCoder) [reply client/src/components/profile/components/TimeLine.js:90]: Ok, thanks

**Comment by dipayanDebTheCoder** [comment]
> Nice work. Can we possibly add/update tests?

I have added the tests for both Certificatesettings and Timeline

**Review by moT01 — CHANGES_REQUESTED** [review]
Screen Shot 2020-11-03 at 9 45 24 AM

Here's a screen shot of these changes. The `view` button shows up on all timeline items. It should not show up for regular challenges or certifications (pretty sure). Also, I would prefer the `view all` to just say `view` with the little drop down - my opinion. I am fine with this style implementation, but here's a side thought... What about instead of a `solution` column with a `view` button next to all of the projects - we add an icon next to all the projects like we did for the certifications, and then clicking title/item would take you to the solution. This doesn't really leave room for a source code/github link - but I think it would look pretty slick. Like I said, I am fine with this implementation as well.

**Comment by dipayanDebTheCoder** [comment]
> Screen Shot 2020-11-03 at 9 45 24 AM
> 
> Here's a screen shot of these changes. The `view` button shows up on all timeline items. It should not show up for regular challenges or certifications (pretty sure). 

You mean to say that it should only show up for projects ? Am I correct ?

**Comment by moT01** [comment]
Yes. When I click on the `view` button next to one of the regular challenges, it doesn't do anything. Unless I'm missing something - or there's a plan for it or something.

**Comment by dipayanDebTheCoder** [comment]
> Yes. When I click on the `view` button next to one of the regular challenges, it doesn't do anything. Unless I'm missing something - or there's a plan for it or something.



> Yes. When I click on the `view` button next to one of the regular challenges, it doesn't do anything. Unless I'm missing something - or there's a plan for it or something.
> 
> Although, nobody else mentioned it - so now I'm doubting myself.

Chain search query helper to narrow search results.  This is a regular challenge. But I believe when  you check the view button a link should open. Can you check once

**Comment by moT01** [comment]
Nov-03-2020 13-50-20

I tried to get one of each type of challenge on my timeline - In order from top to bottom, here's what I found:
| Order in gif |            Title           |               Challenge Type              |                                Status                               |                                    Solution                                   |
|--------------|:--------------------------:|:-----------------------------------------:|:-------------------------------------------------------------------:|:-----------------------------------------------------------------------------:|
| 1            |    Legacy Data Vis Cert    |               Certification               | View button does nothing - clicking the title takes you to the cert |           Remove button - or make the button link to the certificate          |
| 2            | Metric-Imperial Convert    | Back-end Project with two URL's submitted | Everything works - button labelled `view all`                       | This is fine - but I would prefer the button say `view` instead of `view all` |
| 3            | Meet the Node console      | Back-end Challenge                        | View button does nothing                                            | Remove button                                                                 |
| 4            | Roman Numeral Converter    | Javascript Project                        | Everything works                                                    |                                                                               |
| 5            | Say Hello to HTML Elements | Front-end Challenge                       | View button does nothing                                            | Remove button                                                                 |
| 6            | Map Data Across the Globe  | Front-end Project                         | Everything works                                                    |                                                                               |
| Not pictured | Any back-end project       | Back-end Project with one URL submitted   | Everything works (the button is labelled `view`)                    |                                                                               |

Checked in firefox and Chrome

**Comment by dipayanDebTheCoder** [comment]
> Checked in firefox and Chrome

I believe I have made the required changes. However the commit was just before you last comment. So I am requesting a review. Thanks for all the help.
proof

**Comment by dipayanDebTheCoder** [comment]
> Nov-03-2020 13-50-20
> 
> I tried to get one of each type of challenge on my timeline - In order from top to bottom, here's what I found:
> 
> Order in gif	Title	Challenge Type	Status	Solution
> 1	Legacy Data Vis Cert	Certification	View button does nothing - clicking the title takes you to the cert	Remove button - or make the button link to the certificate
> 2	Metric-Imperial Convert	Back-end Project with two URL's submitted	Everything works - button labelled `view all`	This is fine - but I would prefer the button say `view` instead of `view all`
> 3	Meet the Node console	Back-end Challenge	View button does nothing	Remove button
> 4	Roman Numeral Converter	Javascript Project	Everything works	
> 5	Say Hello to HTML Elements	Front-end Challenge	View button does nothing	Remove button
> 6	Map Data Across the Globe	Front-end Project	Everything works	
> Not pictured	Any back-end project	Back-end Project with one URL submitted	Everything works (the button is labelled `view`)	
> Checked in firefox and Chrome

Hi, I have modified the code to make sure only project's have the view button. 
Is there anything more to do in this ? If this PR is mergeable then I will move onto another issue. Thanks for your help.

**Review by moT01 — APPROVED** [review]
LGTM @dipayanDebTheCoder 🎉 Thanks for the quick updates to this and congratulations on your first contribution to freeCodeCamp 🎉 There's plenty more issues, we welcome your help with any of them 😄

**Pull request merged**